ui/components/app/assets/nfts/nft-details/nft-details.tsx

```diff
@@ -165,6 +165,19 @@ export default function NftDetails({ nft }: { nft: Nft }) {
   };
 
   const { chainId } = currentChain;
+
+  useEffect(() => {
+    const getAssetImageUrl = async () => {
+      const assetImageUrl = await getAssetImageURL(
+        imageOriginal ?? image,
+        ipfsGateway,
+      );
+      setAssetImageUrl(assetImageUrl);
+    };
+
+    getAssetImageUrl();
+  }, []);
```

Comment:
Don't need to make this change, but I noticed this logic is repeated in a couple places. I wonder if this would be a good candidate for a react hook?

Reply:
Refactored code in [420a435](https://github.com/MetaMask/metamask-extension/pull/26984/commits/420a4350c2bbd619c4f04a6055673390e9796572) 🙏